lib/src/backend.rs

```diff
@@ -194,6 +195,24 @@ pub struct CopyRecord {
     pub source_commit: CommitId,
 }
 
+/// Describes the copy history of a file. The copy object is unchanged when a
+/// file is modified.
+#[derive(ContentHash, Debug, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct CopyHistory {
+    /// The file's current path.
+    pub current_path: RepoPathBuf,
+    /// IDs of the files that became the current incarnation of this file.
+    ///
+    /// A newly created file has no parents. A regular copy or rename has one
+    /// parent. A merge of multiple files has multiple parents.
+    pub parents: Vec<CopyId>,
+    /// An optional piece of data to give the Copy object a different ID. May be
+    /// randomly generated. This allows a commit to say that a file was replaced
+    /// by a new incarnation of it, indicating a logically distinct file
+    /// taking the place of the previous file at the path.
+    pub salt: Vec<u8>,
```

Comment:
Maybe this can be `enum salt|parents`, but `salt, parents` might be handy as low-level representation.